qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,227 @@
+            netperf_test_duration = params.get_numeric("netperf_test_duration")
+            test_protocols = params.get("test_protocols", "TCP_STREAM")
+            netperf_output_unit = params.get("netperf_output_unit")
+            test_option = params.get("test_option", "")
+            test_option += " -l %s" % netperf_test_duration
+            if netperf_output_unit in "GMKgmk":
+                test_option += " -f %s" % netperf_output_unit
+            t_option = "%s -t %s" % (test_option, test_protocols)
+            n_client.bg_start(utils_net.get_host_ip_address(params),
+                              t_option,
+                              params.get_numeric("netperf_para_sessions"),
+                              params.get("netperf_cmd_prefix", ""),
+                              package_sizes=params.get("netperf_sizes"))
+            if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 1,
+                                   "Wait netperf test start"):
+                logging.info("Netperf test start successfully.")
+            else:
+                test.error("Can not start netperf client.")
+            utils_misc.wait_for(
+                lambda: not n_client.is_netperf_running(),
+                netperf_test_duration, 0, 5,
+                "Wait netperf test finish %ss" % netperf_test_duration)
+            time.sleep(5)
+        finally:
+            n_server.stop()
+
+    login_timeout = params.get_numeric("login_timeout", 360)
+    netperf_stress = params.get("run_bgstress")
+    netperf_stress_run_flag = params.get("netperf_stress_run_flag")
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_serial_login(timeout=login_timeout)
+    enable_multi_queues(vm)
+    n_client = utils_netperf.NetperfClient(
+        vm.get_address(), params.get("client_path"),
+        netperf_source=os.path.join(data_dir.get_deps_dir("netperf"),
+                                    params.get("netperf_client_link")),
+        client=params.get("shell_client"),
+        username=params.get("username"),
+        password=params.get("password"),
+        compile_option=params.get("compile_option", ""))
+    n_server = utils_netperf.NetperfServer(
+        utils_net.get_host_ip_address(params),
+        params.get("server_path", "/var/tmp"),
+        netperf_source=os.path.join(data_dir.get_deps_dir("netperf"),
+                                    params.get("netperf_server_link")),
+        password=params.get("hostpassword"),
+        compile_option=params.get("compile_option", ""))
+    wait_time = params.get_numeric("wait_bg_time", 60)
+    ping_lost_ratio = params.get_numeric("background_ping_package_lost_ratio", 5)
+    ping_time = params.get_numeric("background_ping_time", 60)
+    guest_ip = vm.get_address()
+    required_reboot = True
+    try:
+        while ping_test:
+            session_serial = vm.wait_for_serial_login(timeout=login_timeout)
+            n_client.session = session_serial
```

Comment:
What's the meaning of this?

Reply:
After the reboot,n_client.session does not change and need to be reassigned.